examples/gcoap/client.c

```diff
@@ -287,7 +345,12 @@ int gcoap_cli_cmd(int argc, char **argv)
         }
 
         if (_proxied) {
-            uri_len = snprintf(proxy_uri, 64, "coap://[%s]:%s%s", argv[apos], argv[apos+1], uri);
+            if (strstr(argv[apos], ":") != NULL) { /* IPv6 might contain '.', but IPv4 cannot contain ':' */
+                uri_len = snprintf(proxy_uri, 64, "coap://[%s]:%s%s", argv[apos], argv[apos+1], uri);
```

Comment:
```suggestion
                uri_len = snprintf(proxy_uri, 64, "coap://[%s]:%s%s",
                                   argv[apos], argv[apos + 1], uri);
```

Reply:
Exceeded the limit by one character :( 

I changed all lines as suggested